mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -316,10 +317,11 @@ class BackupNotifier extends StateNotifier<BackUpState> {
     }
 
     // Find asset that were backup from selected albums
+    var assetBox = Hive.box<HiveBackupAsset>(backupAssetInfoBox);
     Set<String> selectedAlbumsBackupAssets =
         Set.from(allUniqueAssets.map((e) => e.id));
     selectedAlbumsBackupAssets
-        .removeWhere((assetId) => !allAssetsInDatabase.contains(assetId));
+        .removeWhere((assetId) => !allAssetsInDatabase.contains(assetId) && !assetBox.containsKey(assetId));
```

Comment:
@alextran1502 if we marked upload files, do we need fetch all assets from db here?

Reply:
I think we have a misunderstand here about the implementation. 

I think we want to "mark" the duplicated file in the HiveBox, not the uploaded file. The number of duplicated file will always be lower than the number of uploaded file. So if the user reinstall the app, the number of collision is lower if we don't have HiveBox that store the duplicated file. 

Are we on the same page here?

Reply:
For **the number of collision**, I still check the existed files with `deviceAssetId`, so we don't need re-upload again if the user reinstalls the app.

I store all uploaded files to the hive box because I wanna replace `getAllAssetInDb` API. The problem I care about is the memory will increase. If you have other plans instead, I'll just save the duplicates.

Reply:
Why do you want to replace the `getAllAssetInDb`?

Reply:
to reduce memory, time when we start backup. Loading all data from db is bad behavior. If you have plan for pagination or another to avoid this issue. We can keep old implement (fetch all asset in db) until we have the new implement.

Reply:
I thought about that, so the method of `getAllAssetInDb` return an array of just string of the `deviceAssetId` for that specific device, not the whole collection from other devices. I think this way we always have the latest info about the asset count on the device.

Reply:
hmm, if you thought about it and you're ok with it, then I don't think it's a problem for me. The final solution is asset will only keep duplicate files. Are you ok?

Reply:
Let's keep the current implementation and only saved duplicated files Id.